src/wp-includes/robots-template.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Robots template functions.
+ *
+ * @package WordPress
+ * @subpackage Robots
+ * @since 5.7.0
+ */
+
+/**
+ * Displays the robots meta tag as necessary.
+ *
+ * Gathers robots directives to include for the current context, using the
+ * {@see 'wp_robots'} filter. The directives are then sanitized, and the
+ * robots meta tag is output if there is at least one relevant directive.
+ *
+ * @since 5.7.0
+ */
+function wp_robots() {
+	/**
+	 * Filters the directives to be included in the 'robots' meta tag.
+	 *
+	 * The meta tag will only be included as necessary.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param array $robots Associative array of directives. Every key must be the name of the directive, and the
+	 *                      corresponding value must either be a string to provide as value for the directive or a
+	 *                      boolean `true` if it is a boolean directive, i.e. without a value.
+	 */
+	$robots = apply_filters( 'wp_robots', array() );
+
+	// Don't allow mutually exclusive directives.
+	if ( ! empty( $robots['follow'] ) ) {
+		unset( $robots['nofollow'] );
+	}
+	if ( ! empty( $robots['nofollow'] ) ) {
+		unset( $robots['follow'] );
+	}
+
+	$robots_strings = array();
+	foreach ( $robots as $directive => $value ) {
+		if ( is_string( $value ) ) {
+			// If a string value, include it as value for the directive.
+			$robots_strings[] = "{$directive}:{$value}";
+		} elseif ( $value ) {
+			// Otherwise, include the directive if it is truthy.
+			$robots_strings[] = $directive;
+		}
+	}
+
+	if ( empty( $robots_strings ) ) {
+		return;
+	}
+
+	echo "<meta name='robots' content='" . esc_attr( implode( ', ', $robots_strings ) ) . "' />\n";
+}
+
+/**
+ * Adds noindex to the robots meta tag if required by the site configuration.
+ *
+ * If a blog is marked as not being public then noindex will be output to
+ * tell web robots not to index the page content. Add this to the
+ * {@see 'wp_robots'} filter.
+ *
+ * Typical usage is as a {@see 'wp_robots'} callback:
+ *
+ *     add_filter( 'wp_robots', 'wp_robots_noindex' );
+ *
+ * @since 5.7.0
+ * @see wp_robots_no_robots()
+ *
+ * @param array $robots Associative array of robots directives.
+ * @return array Filtered robots directives.
+ */
+function wp_robots_noindex( array $robots ) {
```

Comment:
Maybe we talked about this before, but I think the `array` type should not be specified here. Otherwise, if any other preceding `wp_robots` filter returns anything other than an array, a fatal error will ensue, no?